src/main/java/com/meta/cp4m/message/WAMessageHandler.java

```diff
@@ -257,4 +284,36 @@ private void markRead(Identifier phoneNumberId, String messageId) {
       LOGGER.error("unable to mark message as read", e);
     }
   }
+
+  private String getUrlFromID(String mediaID) throws IOException {
+     String graphAPI = "https://graph.facebook.com/v20.0/";
+   try {
+
+     Content content = Request.get(graphAPI + mediaID)
+             .setHeader("Authorization", "Bearer " + accessToken)
+             .setHeader("appsecret_proof", appSecretProof)
+             .execute().returnContent();
+
+     String jsonResponse = content.asString();
+     ObjectMapper objectMapper = new ObjectMapper();
+     JsonNode jsonNode = objectMapper.readTree(jsonResponse);
+     return jsonNode.get("url").asText();
+   } catch (IOException e) {
+     LOGGER.error("Unable to retrieve media URL from ID", e);
+     return "";
+   }
+  }
+
+  private Content getMediaFromUrl(String url) throws IOException {
+    try {
+      Content media = Request.get(url)
+              .setHeader("Authorization", "Bearer " + accessToken)
+              .setHeader("appsecret_proof", appSecretProof)
+              .execute().returnContent();
+      return media;
+    } catch (IOException e) {
```

Comment:
just throw this exception don't log it, you already declare it in the method signature

Reply:
you're catching `IOException` but IOException is in the method declaration, don't catch it